### PR TITLE
14 The Document Object Model

### PR DESCRIPTION
**Pull request by magdalena-u**
Chapter 14: The Document Object Model:
14_1 Build a table
14_2 Elements by tag name
14_3 The cat's hat

### CODE REVIEW

File: 14_2_elements.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+function byTagName(node, tagName) {
+    const array = [];
+    const elements = node.querySelectorAll(tagName);
+    elements.forEach(element => {
```
Comment (michal-lepik) [inline 14_2_elements.js]: Instead of creating an array and mutating it, try using `map` - it returns new array :)

File: 14_3_cat.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+let cat = document.querySelector('img');
```
Comment (michal-lepik) [inline 14_3_cat.js]: Try using `const` whenever it is possible

File: 14_1_table.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    { name: 'Popocatepetl', height: 5465, place: 'Mexico' },
+    { name: 'Mont Blanc', height: 4808, place: 'Italy/France' },
+];
+
+const div = document.getElementById('mountains');
+const table = document.createElement('table');
+
+div.appendChild(table);
+table.innerHTML += `<tr>
+<th>${Object.keys(MOUNTAINS[0])[0]}</th>
```
Comment (michal-lepik) [inline 14_1_table.js]: I'd move `Object.keys(MOUNTAINS[0])` to separate variable and then use `forEach` to generate that table

File: 14_2_elements.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+function byTagName(node, tagName) {
+    const elements = node.querySelectorAll(tagName);
+    return [...elements].map(el => el.nodeName.toLowerCase());
```
Comment (mateusz-bielak) [inline 14_2_elements.js:3]: You don't have to destructure `elements` in this case. You are using `.map` which returns a new array, so you can do just `elements.map(....`.

File: 14_2_elements.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+}
+
+console.log(byTagName(document.body, 'h1').length);
+// → 1
+console.log(byTagName(document.body, 'span').length);
+// // → 3
+let para = document.querySelector('p');
```
Comment (mateusz-bielak) [inline 14_2_elements.js]: You can use `const` here. And, what `para` means? If it is `paragraph`, then please, use full name to omit any misunderstandings.

File: 14_3_cat.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+function animate(time, lastTime) {
+    if (lastTime != null) {
+        angle += (time - lastTime) * 0.001;
+    }
+
+    div.style.top = Math.sin(angle) * 20 + 50 + 'px';
+    div.style.left = Math.cos(angle) * 200 + 250 + 'px';
+
+    hat.style.top = Math.cos(angle) * 10 - 30 + 'px';
+    hat.style.left = Math.sin(angle) * 20 + 80 + 'px';
```
Comment (mateusz-bielak) [inline 14_3_cat.js]: What are all these values like `20`, `50`, `10`, `30` etc? Maybe put those into some descriptive variables?

File: index-DOM.html [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+<html lang="en">
+    <head>
+        <meta charset="UTF-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+        <script defer src="cat.js"></script>
+        <style>
+            /* Table */
+
+            /* body {
```
Comment (mateusz-bielak) [inline index-DOM.html:11]: Do you need all that commented code?

File: index-DOM.html [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+                height: 80px;
+            }
+        </style>
+        <title>Document</title>
+    </head>
+    <body>
+        <!-- Table -->
+        <!-- <h1>Mountains</h1>
+        <div id="mountains"></div> -->
+        <!-- Elements -->
```
Comment (mateusz-bielak) [inline index-DOM.html:44]: And these comments?

**Review by mateusz-bielak — APPROVED** [review]

**Pull request merged**